pkg/machines/libvirt.es6

```diff
@@ -522,15 +605,20 @@ function spawnVirshReadOnly({connectionName, method, name, failHandler}) {
     return spawnVirsh({connectionName, method, args: ['-r', method, name], failHandler});
 }
 
-function parseDumpxml(dispatch, connectionName, domXml) {
+function getDomainElem(domXml) {
     const xmlDoc = $.parseXML(domXml);
 
     if (!xmlDoc) {
         console.warn(`Can't parse dumpxml, input: "${domXml}"`);
         return ;
     }
 
-    const domainElem = xmlDoc.getElementsByTagName("domain")[0];
+    return xmlDoc.getElementsByTagName("domain")[0];
+}
+
+function parseDumpxml(dispatch, connectionName, domXml) {
+
+    const domainElem = getDomainElem(domXml);
     const osElem = domainElem.getElementsByTagName("os")[0];
```

Comment:
What if the domXml can't be parsed? 
It will cause error by calling `undefined.getElementsByTagName()`